File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.java
import java.util.*;
class Solution {
    public int solution(String s) {
        HashMap<String, String> num = new HashMap<>();
        num.put("zero", "0");
        num.put("one", "1");
        num.put("two", "2");
        num.put("three", "3");
        num.put("four", "4");
        num.put("five", "5");
        num.put("six", "6");
        num.put("seven", "7");
        num.put("eight", "8");
        num.put("nine", "9");

        for(String key : num.keySet()){
            if(s.contains(key)){
                s = s.replace(key,num.get(key));
            }
        }
        
        return Integer.parseInt(s);
    }
}

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    const numMap = {
    "zero": "0",
    "one": "1",
    "two": "2",
    "three": "3",
    "four": "4",
    "five": "5",
    "six": "6",
    "seven": "7",
    "eight": "8",
    "nine": "9"
    };
    
    for(let i in numMap){
        //RegExp는 Regular Expression(정규표현식) 을 만들 때 쓰는 자바스크립트 내장 클래스
        const regex = new RegExp(i, "g");
        s = s.replace(regex, numMap[i]);
    }  
    return Number(s);
}